merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresResultsCellRepository.java

```diff
@@ -290,10 +290,10 @@ private static Optional<State> getSimulationState(
 
     return Optional.of(
         switch (record.state().status()) {
-          case PENDING -> new ResultsProtocol.State.Pending();
-          case INCOMPLETE -> new ResultsProtocol.State.Incomplete();
-          case FAILED -> new ResultsProtocol.State.Failed(record.state().reason());
-          case SUCCESS -> new ResultsProtocol.State.Success(getSimulationResults(connection, record, planId));
+          case PENDING -> new ResultsProtocol.State.Pending(datasetId);
```

Comment:
🔻 There is an unfortunate subtle difference between `simulationDatasetId` and `datasetId`. In this case, I think we want to update `SimulationDatasetRecord` to have a `simulationDatasetId` field, and pull that field out of the `record` variable here.

Reply:
Would this be separate from the `datasetId` field that already exists in the `SimulationDatasetRecord`?

Reply:
Correct, and this is why I say it's unfortunate 😓 

`simulation_dataset` is a join table between the `simulation` table and the `dataset` table. `datasetId` refers to the `id` field on the `dataset` table, while `simulationDatasetId` refers to the `id` field on the `simulation_dataset` table.

Being a join table, the `simulation_dataset` refers to the `dataset` table using a column of its own called `dataset_id`. So, the `SimulationDatasetRecord.datasetId` field comes from the `simulation_dataset.dataset_id` column, which is a foreign key that refers to the `id` column of the `dataset` table.

Let me know if I made things even more confusing.

Reply:
Ahh gotcha, so I'll add `simulation_dataset.id` to the queries in `{Get,Lookup,Create}SimulationDatasetAction` and add that into the `SimulationDatasetRecord` as `simulationDatasetId`